include/SDL3/SDL_gpu.h

```diff
@@ -1156,210 +1317,241 @@ typedef struct SDL_GPUTextureCreateInfo
-    SDL_GPUDepthStencilState depth_stencil_state;
-    SDL_GpuGraphicsPipelineTargetInfo target_info;
-
-    SDL_PropertiesID props;
+    SDL_GPUShader *vertex_shader;                   /**< The vertex shader used by the graphics pipeline. */
+    SDL_GPUShader *fragment_shader;                 /**< The fragment shader used by the graphics pipeline. */
+    SDL_GPUVertexInputState vertex_input_state;     /**< The vertex layout of the graphics pipeline. */
+    SDL_GPUPrimitiveType primitive_type;            /**< The primitive topology of the graphics pipeline. */
+    SDL_GPURasterizerState rasterizer_state;        /**< The rasterizer state of the graphics pipeline. */
+    SDL_GPUMultisampleState multisample_state;      /**< The multisample state of the graphics pipeline. */
+    SDL_GPUDepthStencilState depth_stencil_state;   /**< The depth-stencil state of the graphics pipeline. */
+    SDL_GpuGraphicsPipelineTargetInfo target_info;  /**< Formats and blend modes for the render targets of the graphics pipeline. */
+
+    SDL_PropertiesID props;                         /**< A properties ID for extensions. Should be 0 if no extensions are needed. */
 } SDL_GPUGraphicsPipelineCreateInfo;
 
+/**
+ * A structure specifying the parameters of a compute pipeline state.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_CreateGPUComputePipeline
+ */
 typedef struct SDL_GPUComputePipelineCreateInfo
 {
-    size_t code_size;
-    const Uint8 *code;
-    const char *entrypoint;
-    SDL_GPUShaderFormat format;
-    Uint32 num_readonly_storage_textures;
-    Uint32 num_readonly_storage_buffers;
-    Uint32 num_writeonly_storage_textures;
-    Uint32 num_writeonly_storage_buffers;
-    Uint32 num_uniform_buffers;
-    Uint32 threadcount_x;
-    Uint32 threadcount_y;
-    Uint32 threadcount_z;
-
-    SDL_PropertiesID props;
+    size_t code_size;                       /**< The size in bytes of the compute shader code pointed to. */
+    const Uint8 *code;                      /**< A pointer to compute shader code. */
+    const char *entrypoint;                 /**< A pointer to a null-terminated UTF-8 string specifying the entry point function name for the shader. */
+    SDL_GPUShaderFormat format;             /**< The format of the compute shader code. */
+    Uint32 num_readonly_storage_textures;   /**< The number of readonly storage textures defined in the shader. */
+    Uint32 num_readonly_storage_buffers;    /**< The number of readonly storage buffers defined in the shader. */
+    Uint32 num_writeonly_storage_textures;  /**< The number of writeonly storage textures defined in the shader. */
+    Uint32 num_writeonly_storage_buffers;   /**< The number of writeonly storage buffers defined in the shader. */
+    Uint32 num_uniform_buffers;             /** The number of uniform buffers defined in the shader. */
+    Uint32 threadcount_x;                   /** The number of threads in the X dimension. Only used by Metal. */
```

Comment:
“Only used by Metal” implies that you don’t need to mess with this if you’re not planning to target Apple platforms, but we currently validate that the thread counts are >= 1 on all backends to ensure you don’t accidentally write unportable code. Explicitly mentioning this may steer clients in the wrong direction.

Reply:
You're right, I'll just document that the value should match the one in your shader.